octobot_commons/async_job.py

```diff
@@ -0,0 +1,127 @@
+# pylint: disable=W0703,R0902,R0913
+#  Drakkar-Software OctoBot-Commons
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import asyncio
+
+import time
+
+from octobot_commons.asyncio_tools import wait_asyncio_next_cycle
+from octobot_commons.logging.logging_util import get_logger
+
+
+class AsyncJob:
+    """
+    Async job management
+    """
+
+    NO_DELAY = -1
+
+    def __init__(
+        self,
+        callback,
+        execution_interval_delay=NO_DELAY,
+        min_execution_delay=NO_DELAY,
+        job_dependencies=None,
+    ):
+        self.logger = get_logger(self.__class__.__name__)
+        self.is_running = False
+        self.is_scheduled = False
+        self.last_execution_time = 0
+        self.execution_interval_delay = execution_interval_delay
+        self.min_execution_delay = min_execution_delay
+        self.callback = callback
```

Comment:
We should be able to register multiple callbacks to ensure they are not called simultaneously (to prevent closed order from being triggered at the same time as open ones)

Reply:
this is managed by using multiple jobs and job_dependencies

Reply:
Yep, I realized this after, my comment is irrelevant